adapters/repos/db/shard_dimension_tracking.go

```diff
@@ -36,15 +36,19 @@ func (s *Shard) Dimensions() int {
 }
 
 func (s *Shard) sendVectorDimensionsMetric(count int) {
-	metric, err := s.promMetrics.VectorDimensionsSum.
-		GetMetricWithLabelValues(s.index.Config.ClassName.String(), s.name)
-	if err == nil {
-		metric.Set(float64(count))
+	if s.promMetrics != nil {
+		metric, err := s.promMetrics.VectorDimensionsSum.
+			GetMetricWithLabelValues(s.index.Config.ClassName.String(), s.name)
+		if err == nil {
+			metric.Set(float64(count))
+		}
 	}
 }
 
 func (s *Shard) initDimensionTracking() {
 	if s.index.Config.TrackVectorDimensions {
+		// always send vector dimensions at startup if tracking is enabled
+		s.sendVectorDimensionsMetric(s.Dimensions())
 		// start tracking vector dimensions goroutine only when tracking is enabled
 		go func() {
 			t := time.NewTicker(5 * time.Minute)
```

Comment:
We need to `defer t.Stop()` to avoid leaking resources

Reply:
Fixed